File: src/main/java/itmo/is/project/mapper/module/storage/StoredResourceMapper.java
package itmo.is.project.mapper.module.storage;

import itmo.is.project.dto.module.storage.StoredResourceDto;
import itmo.is.project.mapper.EntityMapper;
import itmo.is.project.model.module.storage.StoredResource;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper
public interface StoredResourceMapper extends EntityMapper<StoredResourceDto, StoredResource> {

    @Override
    @Mapping(source = "storageModule.id", target = "storageModuleId")
    StoredResourceDto toDto(StoredResource entity);
}

File: src/main/kotlin/itmo/isproject/storage/mapper/StoredResourceMapper.kt
package itmo.isproject.storage.mapper

import itmo.isproject.storage.dto.StoredResourceDto
import itmo.isproject.shared.common.mapper.EntityMapper
import itmo.isproject.storage.model.StoredResource
import org.mapstruct.Mapper
import org.mapstruct.Mapping

@Mapper
interface StoredResourceMapper : EntityMapper<StoredResourceDto, StoredResource> {

    @Mapping(source = "storageModule.id", target = "storageModuleId")
    override fun toDto(entity: StoredResource): StoredResourceDto
}
